vehicles/src/modules/permit-application-payment/payment/payment.service.ts

```diff
@@ -91,10 +97,33 @@ export class PaymentService {
     return `${year}${monthPadded}${dayPadded}${hoursPadded}${minutesPadded}`;
   };
 
-  private queryHash = (transaction: Transaction) => {
+  private queryHash = async (transaction: Transaction) => {
     const redirectUrl = process.env.PAYBC_REDIRECT;
     const date = new Date().toISOString().split('T')[0];
 
+    const glCodeDetails = await Promise.all(
+      transaction.permitTransactions.map(
+        async ({ permit: { permitType }, transactionAmount }) => ({
+          glCode: await getFromCache(
+            this.cacheManager,
+            CacheKey.PERMIT_TYPE_GL_CODE,
+            permitType,
+          ),
+          amount: transactionAmount,
+        }),
+      ),
+    );
+
+    const groupedGlCodes = glCodeDetails.reduce((acc, { glCode, amount }) => {
+      acc.set(glCode, (acc.get(glCode) || 0) + amount);
+      return acc;
+    }, new Map<string, number>());
+
+    // Format the output string
+    const revenue = Array.from(groupedGlCodes.entries())
+      .map(([glCode, amount], index) => `${index + 1}:${glCode}:${amount}`)
```

Comment:
Why is it `index + 1`? I think an example in the code comment will make it easier to read.

Reply:
Done